### PR TITLE
[ALLUXIO-2119] Remove use of whitebox in FileInStreamTest

### PR DESCRIPTION
**Pull request by aaudiber**
https://alluxio.atlassian.net/browse/ALLUXIO-2119

### CODE REVIEW

File: core/client/src/main/java/alluxio/client/file/BaseFileSystem.java [hunk 1]
```diff
@@ -262,7 +262,7 @@ public FileInStream openFile(AlluxioURI path, OpenFileOptions options)
       throw new FileNotFoundException(
           ExceptionMessage.CANNOT_READ_DIRECTORY.getMessage(status.getName()));
     }
-    return FileInStream.create(status, options.toInStreamOptions());
+    return FileInStream.create(status, options.toInStreamOptions(), FileSystemContext.INSTANCE);
```
Comment (calvinjia) [inline core/client/src/main/java/alluxio/client/file/BaseFileSystem.java]: Its kind of weird to pass a singleton, will it be refactored later?

File: core/client/src/test/java/alluxio/client/file/FileInStreamTest.java [hunk 2]
```diff
@@ -567,7 +555,7 @@ public void missingLocationPolicyTest() {
     try {
       mTestStream =
           new FileInStream(mStatus, InStreamOptions.defaults().setReadType(ReadType.CACHE)
-              .setLocationPolicy(null));
+              .setLocationPolicy(null), FileSystemContext.INSTANCE);
```
Comment (calvinjia) [inline core/client/src/test/java/alluxio/client/file/FileInStreamTest.java]: Why not use `mContext` as well here?

File: core/client/src/test/java/alluxio/client/file/MockFileInStream.java [hunk 3]
```diff
@@ -29,7 +29,7 @@
    * @param bytes the bytes to supply
    */
   public MockFileInStream(byte[] bytes) {
-    super(new URIStatus(new FileInfo()), InStreamOptions.defaults());
+    super(new URIStatus(new FileInfo()), InStreamOptions.defaults(), FileSystemContext.INSTANCE);
```
Comment (calvinjia) [inline core/client/src/test/java/alluxio/client/file/MockFileInStream.java]: Should this be taking in the context as an argument?

File: core/client/src/main/java/alluxio/client/file/BaseFileSystem.java (hunk 1 above)
Comment (aaudiber) [inline core/client/src/main/java/alluxio/client/file/BaseFileSystem.java]: oh, I didn't realize I'd already dependency injected this class, switched to `mContext` 👍

File: core/client/src/test/java/alluxio/client/file/FileInStreamTest.java (hunk 2 above)
Comment (aaudiber) [inline core/client/src/test/java/alluxio/client/file/FileInStreamTest.java]: done

File: core/client/src/test/java/alluxio/client/file/MockFileInStream.java (hunk 3 above)
Comment (aaudiber) [inline core/client/src/test/java/alluxio/client/file/MockFileInStream.java]: done

**Comment by aaudiber** [comment]
@calvinjia Comments addressed

File: core/client/src/test/java/alluxio/client/file/FileInStreamTest.java [hunk 4]
```diff
@@ -540,23 +535,16 @@ public void skipNegativeTest() throws IOException {
...
-    FileWriteLocationPolicy policy = Whitebox.getInternalState(mTestStream, "mLocationPolicy");
-    Assert.assertTrue(policy instanceof LocalFirstPolicy);
-
-    // configure a different policy
-    mTestStream =
-        new FileInStream(mStatus, InStreamOptions.defaults().setReadType(ReadType.CACHE)
-            .setLocationPolicy(new RoundRobinPolicy()));
-    policy = Whitebox.getInternalState(mTestStream, "mLocationPolicy");
-    Assert.assertTrue(policy instanceof RoundRobinPolicy);
+  public void locationPolicyTest() throws Exception {
```
Comment (calvinjia) [inline core/client/src/test/java/alluxio/client/file/FileInStreamTest.java]: Are we intentionally removing the test for the default case?

Comment (aaudiber) [inline core/client/src/test/java/alluxio/client/file/FileInStreamTest.java]: Yes, the default configuration functionality belongs to `InStreamOptions`, not `FileInStream`

File: core/client/src/main/java/alluxio/client/file/FileInStream.java [hunk 5]
```diff
@@ -120,11 +122,11 @@ public static FileInStream create(URIStatus status, InStreamOptions options) {
    * @param status the file status
    * @param options the client options
    */
-  protected FileInStream(URIStatus status, InStreamOptions options) {
+  protected FileInStream(URIStatus status, InStreamOptions options, FileSystemContext context) {
```
Comment (calvinjia) [inline core/client/src/main/java/alluxio/client/file/FileInStream.java]: In another PR we put the context first, do we want to have a consistent argument ordering?

Comment (aaudiber) [inline core/client/src/main/java/alluxio/client/file/FileInStream.java]: Yes, all contexts will be first. I'm doing a followup PR to make this consistent everywhere

Comment (calvinjia) [inline core/client/src/main/java/alluxio/client/file/FileInStream.java]: Got it.

**Comment by calvinjia** [comment]
LGTM

**Pull request merged**